crates/ra_ide_api_light/src/assists/introduce_variable.rs

```diff
@@ -26,6 +26,9 @@ pub fn introduce_variable<'a>(ctx: AssistCtx) -> Option<Assist> {
             false
         };
         if is_full_stmt {
+            if expr.syntax().kind() == BLOCK_EXPR {
```

Comment:
Yeah, I think this is not exactly correct.... We also should check if expression is an `if`, `match` etc.

I think what we really need to check here is "does the `ExprStmt` from above already has a semicolon"?

To do that, we need to change `is_full_stmt: bool` to `full_stmt: Option<ast::ExprStmt>` and add a `has_semi` method to `ExprStmt` here: https://github.com/rust-analyzer/rust-analyzer/blob/e465032daf0d2594c09ad693b2ae816b96e883b2/crates/ra_syntax/src/ast.rs#L270